client/src/screens/search-groups.screen.js

```diff
@@ -68,21 +69,37 @@ class Group extends Component {
   constructor(props) {
     super(props);
     this.myGroups = this.props.myGroups;
-    this.joinGroupQuery = this.props.joinGroupQuery.bind(this);
-    this.joinGroup = this.props.joinGroup.bind(this, this.props.group);
+    this.joinGroup = this.joinGroup.bind(this);
+    this.groupUpdate = this.props.groupUpdate.bind(this)
+    this.state = {
+      memberAlready: _.some(this.myGroups, g => g.id === this.props.group.id),
```

Comment:
nice!

Reply:
there is one downside to this approach, though... this is only checked at start time which means that if something updates the data and calls `render()` this state will not be updated. For that reason it probably should be moved back to `render()`